crates/optimism/txpool/src/interop.rs

```diff
@@ -0,0 +1,35 @@
+//! Additional support for pooled transactions with [`TransactionInterop`]
+
+/// Helper trait that allows attaching a [`TransactionInterop`].
+pub trait MaybeInteropTransaction {
+    /// Attach a [`TransactionInterop`].
+    fn set_interop(&self, interop: TransactionInterop);
+
+    /// Get attached [`TransactionInterop`] if any.
+    fn interop(&self) -> Option<TransactionInterop>;
+
+    /// Helper that sets the interop and returns the instance again
+    fn with_interop(self, interop: TransactionInterop) -> Self
+    where
+        Self: Sized,
+    {
+        self.set_interop(interop);
+        self
+    }
+}
+
+/// Helper to keep track of cross transaction interop validity
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub struct TransactionInterop {
+    /// Unix timestamp until which tx if considered valid by supervisor.
+    ///
+    /// If None - tx is not validated, it should be automatically revalidated by interop tracker.
+    pub timeout: u64,
```

Comment:
still has to be `AtomicU64` based on what matze mentioned in past review that most of the time txns are passed around reth wrapped in `Arc`

Reply:
We have RwLock now, so we are fine with u64

Reply:
That's how we store this structure:
```
interop: Arc<RwLock<Option<TransactionInterop>>>,
```